BUILDING.md

```diff
@@ -1,43 +1,19 @@
 # Building OpenAssetIO
 
-## Quick start: Python only
-
-For use in a Python-only environment, the Python package can be
-installed from the root of the repository using
-
-```shell
-pip install .
-```
-
-This will automatically build and install the Python extension module,
-along with the Python sources.
-
-> **Note**
->
-> CMake's `find_package` must be able to locate dependencies.
-> Consider using the [`CMAKE_TOOLCHAIN_FILE`](https://cmake.org/cmake/help/v3.24/envvar/CMAKE_TOOLCHAIN_FILE.html)
-> environment variable to augment CMake's package search paths.
->
-> For example: to use `conan` to resolve dependencies, from the root of
-> the repository run (assuming a Linux host, translate as necessary):
->
-> ```shell
-> conan install -if .conan resources/build --build=missing
-> export CMAKE_TOOLCHAIN_FILE=`pwd`/.conan/conan_paths.cmake
-> pip install .
-> ```
-
-This will install a single binary module with the OpenAssetIO core
-library statically linked into it, and so is not suitable for use in a
-C++ application.
-
-For users that wish to modify the Python sources and test their changes
-without reinstalling, [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs)
-are also supported
-
-```shell
-pip install --editable .
-```
+## Contents
+
+- [System requirements](#system-requirements)
+- [Library dependencies](#library-dependencies)
+  - [Build Dependencies](#build-dependencies)
+  - [Test Dependencies](#test-dependencies)
+- [Building](#building)
+  - [Building via pip](#building-via-pip)
+- [CMake Options](#cmake-options)
+- [Running tests](#running-tests)
+  - [Using `ctest`](#using-ctest)
+- [Sandboxed builds](#sandboxed-builds)
+  - [Docker](#docker)
+  - [Vagrant](#vagrant)
 
 ## System requirements
```

Comment:
[consider] I wonder if we should  be a bit more relaxed here, or at least note that these are the system requirements we assume for the following sections, but "MacOS and Windows works too, translate as necessary".

Reply:
Is this just GCC we're talking about here? I wonder if it's possible to list what specific minimum versions of MSVC/XCode you need installed? Do we know?

Reply:
Should we just change it to `C++17 compliant(?) compiler`

Reply:
I'd quite like to list exact versions, I think it helps, even as an afterthought.
What do you think about
`C++17 compliant compiler (GCC 9.3 / MSVC 16.9 / macOS 10.15)`
?
These are CY2022 specs.

Reply:
^ I like that.  Another option is to link to VFX reference platform and say CY22.

Reply:
"or later" as there are no issues building with newer compilers (I'm using clang 14 all the time)?

Reply:
[fbca361](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/fbca36113879859bfdf31fbf38ad773915bad3f5)